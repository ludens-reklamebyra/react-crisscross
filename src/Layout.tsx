import * as React from 'react';
import styled, { css } from 'styled-components';
import MediaQueries from './MediaQueries';
import { LayoutInterface } from './Interfaces';

const Layout = styled.div<LayoutInterface>`
  ${(props: LayoutInterface) =>
    props.container &&
    css`
      display: flex;
      flex-direction: ${props.direction || 'row'};
      justify-content: ${props.justify || 'flex-start'};
      align-items: ${props.alignItems || 'stretch'};
      flex-wrap: ${props.wrap || 'nowrap'};
      ${props.gap &&
        (props.direction === 'column'
          ? css`
              margin-top: -${props.gap}px;
              margin-bottom: -${props.gap}px;
              > * {
                padding-top: ${props.gap}px;
                padding-bottom: ${props.gap}px;
              }
            `
          : css`
              margin-left: -${props.gap}px;
              margin-right: -${props.gap}px;
              > * {
                padding-left: ${props.gap}px;
                padding-right: ${props.gap}px;
              }
            `)};
    `};
  ${(props: LayoutInterface) =>
    props.item &&
    css`
      ${props.xs && `flex-basis: ${(props.xs / 12) * 100}%;`}
      ${props.sm &&
        `@media only screen and (min-width: ${MediaQueries.small}px) {
          flex-basis:  ${(props.sm / 12) * 100}%;
        }`}
      ${props.md &&
        `@media only screen and (min-width: ${MediaQueries.medium}px) {
          flex-basis:  ${(props.md / 12) * 100}%;
        }`}
      ${props.lg &&
        `@media only screen and (min-width: ${MediaQueries.large}px) {
          flex-basis:  ${(props.lg / 12) * 100}%;
        }`}
      ${props.xl &&
        `@media only screen and (min-width: ${MediaQueries.xlarge}px) {
          flex-basis:  ${(props.xl / 12) * 100}%;
        }`}

      ${props.flex &&
        css`
          flex: ${props.flex || '1 1 auto'};
        `};
      ${props.grow &&
        css`
          flex-grow: ${props.grow || 0};
        `};
      ${props.shrink &&
        css`
          flex-shrink: ${props.shrink || 1};
        `};
      box-sizing: border-box;
      align-self: ${props.alignSelf || 'auto'};
    `};
`;

export default Layout;
