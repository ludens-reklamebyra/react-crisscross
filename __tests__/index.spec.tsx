import * as React from 'react';
import { create } from 'react-test-renderer';
import Layout from '../src/index';

describe('Layout', () => {
  it('renders the simplest example of the Layout', () => {
    const layoutComponent = (
      <Layout container>
        <Layout item>hello world</Layout>
      </Layout>
    );

    const tree = create(layoutComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
