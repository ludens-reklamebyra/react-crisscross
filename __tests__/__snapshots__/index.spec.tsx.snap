// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Custom settings with layout provider 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  margin-left: -16px;
  margin-right: -16px;
}

.c0 > * {
  padding-left: 16px;
  padding-right: 16px;
}

.c1 {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  box-sizing: border-box;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

@media only screen and (min-width:480px) {
  .c1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
  }
}

@media only screen and (min-width:747px) {
  .c1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
  }
}

@media only screen and (min-width:960px) {
  .c1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    I'm positioned left
  </div>
  <div
    className="c1"
  >
    I'm positioned right
  </div>
</div>
`;

exports[`Layout Simple two column layout 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  margin-left: -16px;
  margin-right: -16px;
}

.c0 > * {
  padding-left: 16px;
  padding-right: 16px;
}

.c1 {
  box-sizing: border-box;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

@media only screen and (min-width:1280px) {
  .c1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    I'm positioned left
  </div>
  <div
    className="c1"
  >
    I'm positioned right
  </div>
</div>
`;

exports[`Layout Simplest example of the Layout 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.c1 {
  box-sizing: border-box;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    hello world
  </div>
</div>
`;
